Fix API integration tests failing due to an undefined alias

## Description

We acknowledged that the following tests are failing

* test_rbac_black_rootcheck_endpoints.tavern.yaml
* test_rbac_black_syscheck_endpoints.tavern.yaml
* test_rbac_white_rootcheck_endpoints.tavern.yaml
* test_rbac_white_syscheck_endpoints.tavern.yaml

Due to an undefined YAML alias 

```console
(framework) ➜  integration git:(main) ✗ pytest -vv --nobuild test_rbac_white_rootcheck_endpoints.tavern.yaml
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.10.18, pytest-7.3.1, pluggy-1.6.0 -- /home/nstefani/.pyenv/versions/3.10.18/envs/framework/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.10.18', 'Platform': 'Linux-6.17.4-76061704-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.3.1', 'pluggy': '1.6.0'}, 'Plugins': {'asyncio': '0.18.1', 'tavern': '1.23.5', 'trio': '0.8.0', 'anyio': '4.1.0', 'html': '2.1.1', 'metadata': '3.1.1', 'cov': '4.1.0'}}
rootdir: /home/nstefani/git/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, trio-0.8.0, anyio-4.1.0, html-2.1.1, metadata-3.1.1, cov-4.1.0
asyncio: mode=auto
collected 0 items / 1 error

====================================================================================================== ERRORS =======================================================================================================
_________________________________________________________________________ ERROR collecting test_rbac_white_rootcheck_endpoints.tavern.yaml __________________________________________________________________________
/home/nstefani/.pyenv/versions/framework/lib/python3.10/site-packages/tavern/testutils/pytesthook/file.py:313: in collect
    all_tests = list(
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/__init__.py:130: in load_all
    yield loader.get_data()
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/constructor.py:45: in get_data
    return self.construct_document(self.get_node())
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:27: in get_node
    return self.compose_document()
/home/nstefani/.pyenv/versions/framework/lib/python3.10/site-packages/tavern/util/loader.py:38: in compose_document
    node = self.compose_node(None, None)
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:84: in compose_node
    node = self.compose_mapping_node(anchor)
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:133: in compose_mapping_node
    item_value = self.compose_node(node, item_key)
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:82: in compose_node
    node = self.compose_sequence_node(anchor)
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:111: in compose_sequence_node
    node.value.append(self.compose_node(node, index))
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:84: in compose_node
    node = self.compose_mapping_node(anchor)
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:133: in compose_mapping_node
    item_value = self.compose_node(node, item_key)
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:84: in compose_node
    node = self.compose_mapping_node(anchor)
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:133: in compose_mapping_node
    item_value = self.compose_node(node, item_key)
/home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/yaml/composer.py:68: in compose_node
    raise ComposerError(None, None, "found undefined alias %r"
E   yaml.composer.ComposerError: found undefined alias 'permission_denied'
E     in "/home/nstefani/git/wazuh/api/test/integration/test_rbac_white_rootcheck_endpoints.tavern.yaml", line 57, column 11
================================================================================================= warnings summary ==================================================================================================
../../../../../.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/_pytest/nodes.py:642
  /home/nstefani/.pyenv/versions/3.10.18/envs/framework/lib/python3.10/site-packages/_pytest/nodes.py:642: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().from_parent(parent=parent, fspath=fspath, path=path, **kw)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================================== short test summary info ==============================================================================================
ERROR test_rbac_white_rootcheck_endpoints.tavern.yaml - yaml.composer.ComposerError: found undefined alias 'permission_denied'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================================================ 1 warning, 1 error in 0.29s ============================================================================================
```

This issue aims to fix the mentioned tests and get them running again

--- a/api/test/integration/test_rbac_black_rootcheck_endpoints.tavern.yaml
+++ b/api/test/integration/test_rbac_black_rootcheck_endpoints.tavern.yaml
@@ -55,4 +55,6 @@ stages:
       params:
         agents_list: '001'
     response:
-      <<: *permission_denied
+      status_code: 403
+      json:
+        error: 4000

--- a/api/test/integration/test_rbac_black_syscheck_endpoints.tavern.yaml
+++ b/api/test/integration/test_rbac_black_syscheck_endpoints.tavern.yaml
@@ -19,7 +19,7 @@ stages:
           failed_items: []
           total_affected_items: 5
           total_failed_items: 0
-        
+
   - name: Try to run a syscheck scan on a list of agents
     request:
       verify: False
@@ -44,7 +44,7 @@ stages:
                 - '003'
           total_affected_items: 2
           total_failed_items: 3
-        
+
   - name: Try to run a syscheck scan on a list of agents
     request:
       verify: False
@@ -55,4 +55,6 @@ stages:
       params:
         agents_list: '001'
     response:
-      <<: *permission_denied
+      status_code: 403
+      json:
+        error: 4000

--- a/api/test/integration/test_rbac_white_rootcheck_endpoints.tavern.yaml
+++ b/api/test/integration/test_rbac_white_rootcheck_endpoints.tavern.yaml
@@ -54,4 +54,9 @@ stages:
       params:
         agents_list: '004'
     response:
-      <<: *permission_denied
+      status_code: 403
+      json:
+        error: 4000
+        dapi_errors:
+          unknown-node: # No permission to see node
+            error: !anystr

--- a/api/test/integration/test_rbac_white_syscheck_endpoints.tavern.yaml
+++ b/api/test/integration/test_rbac_white_syscheck_endpoints.tavern.yaml
@@ -19,7 +19,7 @@ stages:
           failed_items: []
           total_affected_items: 4
           total_failed_items: 0
-        
+
   - name: Try to run a syscheck scan on a list of agents
     request:
       verify: False
@@ -43,7 +43,7 @@ stages:
                 - '008'
           total_affected_items: 3
           total_failed_items: 2
-        
+
   - name: Try to run a syscheck scan on a list of agents
     request:
       verify: False
@@ -54,4 +54,9 @@ stages:
       params:
         agents_list: '004'
     response:
-      <<: *permission_denied
+      status_code: 403
+      json:
+        error: 4000
+        dapi_errors:
+          unknown-node: # No permission to see node
+            error: !anystr

